Xamarin.PropertyEditing.Mac/Controls/BaseEditorControl.cs

```diff
@@ -67,9 +105,22 @@ protected void SetErrors (IEnumerable errors)
 		{
 			errorList = errors;
 
-			errorButton.Enabled = errors != null;
+			actionButton.Enabled = errors != null;
+
 			// Using NSImageName.Caution for now, we can change this later at the designers behest
-			errorButton.Image = errorButton.Enabled ? NSImage.ImageNamed (NSImageName.Caution) : null;
+			actionButton.Image = actionButton.Enabled ? NSImage.ImageNamed ("action-warning-16") : null;
 		}
+
+		void NotifyPropertyButtonClicked ()
+		{
+			PropertyButtonClicked?.Invoke (this, EventArgs.Empty);
+		}
+
+		void NotifyActioButtonClicked ()
+		{
+			ActionButtonClicked?.Invoke (this, EventArgs.Empty);
+		}
+
+		protected abstract void RestDefaultValues ();
```

Comment:
What is `RestDefaultValues` ?

Reply:
@ermau This was meant to hook into the property event when the user decides to reset to that property's default value. I will remove this as the event is there, but would like to talk to you about the intricacies of the implementation for the various states.